aspnetcore/signalr/streaming.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Use streaming in ASP.NET Core SignalR
+author: rachelappel
+description: 
+manager: wpickett
+monikerRange: '>= aspnetcore-2.1'
+ms.author: rachelap
+ms.custom: mvc
+ms.date: 06/07/2018
+ms.prod: aspnet-core
+ms.technology: aspnet
+ms.topic: article
+uid: signalr/streaming
+---
+
+# Use streaming in ASP.NET Core SignalR
+
+By [Brennan Conroy](https://github.com/BrennanConroy)
+
+Streaming is useful for scenarios when you don't have all the data to send to the client yet, but will get it over time and want to update the client whenever new data is available.
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/live/aspnetcore/signalr/streaming/sample) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+## Set up the hub
+
+A hub method automatically becomes a streaming hub method when it returns a variant of `ChannelReader`. For example, return values of `ChannelReader<int>` and `Task<ChannelReader<string>>` are streaming methods. Below is a sample that shows the basics of streaming data to the client. Everytime data is written to the `ChannelReader` the client is notified of the new data. At the end, the `ChannelReader` is completed to tell the client the stream is closed.
+
+[!code-javascript[Streaming hub method](streaming/sample/hubs/streamhub.cs?range=10-29)]
+
+## .NET client
+
+`StreamAsChannelAsync` on the `HubConnection` calls streaming methods on the hub. Pass the hub method name, and arguments defined in the hub method to `StreamAsChannelAsync`. Set the generic in `StreamAsChannelAsync` based on the type returned from the streaming method. A `ChannelReader<T>` is returned from the stream invocation, and represents the stream on the client. To read data, a common pattern is to loop over `WaitToReadAsync` and call `TryRead` when data is available. The loop will end when the stream has been closed by the server, or the cancellation token passed to `StreamAsChannelAsync` is canceled.
```

Comment:
`StreamAsChannelAsync` on the `HubConnection` calls streaming methods on the hub.

->

The `StreamAsChannelAsync` method on `HubConnection` is used to invoke a streaming method.

Reply:
Set the generic in `StreamAsChannelAsync` based on the type returned from the streaming method.

->

The generic parameter to `StreamAsChannelAsync` specifies the type of objects returned by the streaming method.